src/odb/src/db/dbIsolation.cpp

```diff
@@ -0,0 +1,285 @@
+  stream << obj._power_domain;
+  // User Code Begin <<
+  // User Code End <<
+  return stream;
+}
+
+_dbIsolation::~_dbIsolation()
+{
+  if (_name)
+    free((void*) _name);
+  // User Code Begin Destructor
+  // User Code End Destructor
+}
+
+// User Code Begin PrivateMethods
+// User Code End PrivateMethods
+
+////////////////////////////////////////////////////////////////////
+//
+// dbIsolation - Methods
+//
+////////////////////////////////////////////////////////////////////
+
+const char* dbIsolation::getName() const
+{
+  _dbIsolation* obj = (_dbIsolation*) this;
+  return obj->_name;
+}
+
+const char* dbIsolation::getAppliesTo() const
+{
+  _dbIsolation* obj = (_dbIsolation*) this;
+  return obj->_applies_to;
+}
+
+const char* dbIsolation::getClampValue() const
+{
+  _dbIsolation* obj = (_dbIsolation*) this;
+  return obj->_clamp_value;
+}
+
+const char* dbIsolation::getIsolationSignal() const
+{
+  _dbIsolation* obj = (_dbIsolation*) this;
+  return obj->_isolation_signal;
+}
+
+const char* dbIsolation::getIsolationSense() const
+{
+  _dbIsolation* obj = (_dbIsolation*) this;
+  return obj->_isolation_sense;
+}
+
+const char* dbIsolation::getLocation() const
+{
+  _dbIsolation* obj = (_dbIsolation*) this;
+  return obj->_location;
+}
+
+void dbIsolation::setPowerDomain(dbPowerDomain* power_domain)
+{
+  _dbIsolation* obj = (_dbIsolation*) this;
+
+  obj->_power_domain = power_domain->getImpl()->getOID();
+}
+
+dbPowerDomain* dbIsolation::getPowerDomain() const
+{
+  _dbIsolation* obj = (_dbIsolation*) this;
+  if (obj->_power_domain == 0)
+    return NULL;
+  _dbBlock* par = (_dbBlock*) obj->getOwner();
+  return (dbPowerDomain*) par->_powerdomain_tbl->getPtr(obj->_power_domain);
+}
+
+// User Code Begin dbIsolationPublicMethods
+dbIsolation* dbIsolation::create(dbBlock* block, const char* name)
+{
+  _dbBlock* _block = (_dbBlock*) block;
+  if (_block->_isolation_hash.hasMember(name))
+    return nullptr;
+  _dbIsolation* iso = _block->_isolation_tbl->create();
+  iso->_name = strdup(name);
+  ZALLOCATED(iso->_name);
+  _block->_isolation_hash.insert(iso);
+  return (dbIsolation*) iso;
+}
+
+void dbIsolation::destroy(dbIsolation* iso)
+{
+  // TODO
+}
+
+void dbIsolation::setAppliesTo(const char* applies_to){
+  _dbIsolation* obj = (_dbIsolation*) this;
+  obj->_applies_to = strdup(applies_to);
+}
+
+void dbIsolation::setClampValue(const char* clamp_value){
+  _dbIsolation* obj = (_dbIsolation*) this;
+  obj->_clamp_value = strdup(clamp_value);
+}
+
+void dbIsolation::setIsolationSignal(const char* isolation_signal){
+  _dbIsolation* obj = (_dbIsolation*) this;
+  obj->_isolation_signal = strdup(isolation_signal);  
+}
+
+void dbIsolation::setIsolationSense(const char* isolation_sense){
+  _dbIsolation* obj = (_dbIsolation*) this;
+  obj->_isolation_sense = strdup(isolation_sense);
+}
+
+void dbIsolation::setLocation(const char* location){
+  _dbIsolation* obj = (_dbIsolation*) this;
+  obj->_location = strdup(location);
+}
```

Comment:
the use of `strdu` seems like it would leak if these values get set again

Reply:
Using std::string would be better